### DISCUSSION TITLE
How to config the Search feature in the BBN Fusion?

### DISCUSSION THREAD

**Discussion by QXgu** [BlueBrain/nexus#2763] [Q&A] [answered]
Hi,

I have a question about the configuration of the Search feature in the BBN Fusion. The Search page indicates the message that "No Search Config found in project VRE_testing/VREDatasets". I go to the corresponding project and create the resource with the type of "nxv: SearchConfig" but still can not make the search works on the BBN we deployed. May I know what is the correct process to enable the Search SubApp?

Thanks,
Amy

**Comment by dhaneshnm** [comment, marked as answer]
Hi Amy,
Do you still see "No Search Config found in project VRE_testing/VREDatasets" after creaeting the resource? If that is the case, you can try updating the type of the resource to "https://bluebrain.github.io/nexus/vocabulary/SearchConfig" instead of "nxv:SearchConfig". In order for the prefix nxv to expand properly, you need the prefix "nxv" added in the project "API mappings". otherwise, we will have to give the full uri for the type as suggested above.

If you still don't see the search working, then the  user may not have access to this project/resource.

**Reply by QXgu** [reply]
@dhaneshnm Thank you for your help! I am able to see the search without "No Search Config found in project VRE_testing/VREDatasets" after creating the resource with the prefix.
However, after the search page was configured, the search page indicates that "no data" was found while I actually created the resources in the projects. Here is the payload I provided for the SearchConfig resource. May I know why I can not found any data on the Search page?
{ "@context": { "nxv": "https://bluebrain.github.io/nexus/vocabulary/" }, "@id": "Searchconfig", "@type": [ "nxv:SearchConfig" ], "label": "MINDS", "view": "https://my-nexus-deployment.com/v1/views/VRE_testing/VREDatasets/my-view-id" }

**Reply by dhaneshnm** [reply]
@QXgu  The search will query the view  in the 'view' filed in the config. This must be an Elastic search view. Can you see if the view you have added in the 'view' field can be queried?  And it returns data? You can do this by using elastic search query editor in admin. Elastic search query editor can be accessed from "Project tools" in the project view.  
(Screen shot below)

![Screenshot 2021-09-07 at 10 28 15](https://user-images.githubusercontent.com/1479155/132311785-d2f0cd0a-04f9-4e2d-8117-72f4d6bf2e0a.png)

If you can see data in  Elastic search query editor for the view you have added in the search config, but still do not see any data in search, we can inspect the browser network tab to see what is wrong with the query.

**Reply by QXgu** [reply]
@dhaneshnm Thanks. I have tried to query on the view that I was added to but still can not find any resources on the Search page. However, in the Elastic search query editor, I can see and query the resources. May I know whether there are any other solutions to solve this problem?

**Reply by dhaneshnm** [reply]
> {"sort":[{"_createdAt":{"order":"desc"}}],"size":20,"from":0,"track_total_hits":1000000,"query":{"bool":{"filter":{"bool":{"must":[{"match_all":{}},{"term":{"_deprecated":false}}]}}}}}

This is the query that search runs by defualt on the view. If this returns results in the Elastic search query editor, it should return the same result on the search.  You can look in browser the developer tools/network tab to confirm to see if that is indeed the case.

**Reply by QXgu** [reply]
@dhaneshnm I have run this query in the view editor and it can return resources but the Search Page still not working. I found out that after I setting up my SearchConfig resources as shown below, the fields of the table in the Search still remain the same as the default. Is there anything wrong with the content of the SearchConfig resource?
`{
  "@id": "SearchConfig",
  "@type": [
    "https://bluebrain.github.io/nexus/vocabulary/SearchConfig"
  ],
  "fields": [
    {
      "dataIndex": "@type",
      "displayIndex": 0,
      "key": "@type",
      "sortable": true,
      "title": "Type1"
    }
  ],
  "view": "https://my-nexus-deployment.com/v1/views/VRE_testing/VREDatasets/https%3A%2F%2Fbluebrain.github.io%2Fnexus%2Fvocabulary%2FdefaultElasticSearchIndex",
  "label": "SearchConfig",
  "description": "global dataset search"
}`
Also, I found out that when I going to the Search pages, the APIs that run does not contain the _search API. Here is all the APIs run for the Search page. Is there anything wrong here that causes the problem?
![image](https://user-images.githubusercontent.com/86449881/132524464-f38f3c3f-e5c4-4370-8452-979b59b4663e.png)
